hudi-spark/src/main/java/org/apache/hudi/keygen/TimestampBasedKeyGenerator.java

```diff
@@ -88,6 +86,13 @@ public TimestampBasedKeyGenerator(TypedProperties config) {
 
   @Override
   public HoodieKey getKey(GenericRecord record) {
+    String recordKey = getRecordKey(record);
+    String partitionPath = getPartitionPath(record, partitionPathField);
+    return new HoodieKey(recordKey, partitionPath);
+  }
+
+  @Override
+  public String getPartitionPath(GenericRecord record, String partitionPathField) {
```

Comment:
ok.. this is just overriding from `SimpleKeyGenerator` . ack

Reply:
Yeah.